thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -105,40 +117,66 @@ public void run() {
     Preconditions.checkNotNull(configuration.getString(FUNCTION_NAME));
     Preconditions.checkNotNull(configuration.getString(COLLECTION_NAME));
     Preconditions.checkNotNull(configuration.getString(METRIC_NAME));
-    Preconditions.checkNotNull(configuration.getString(CRON_EXPRESSION));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_SIZE));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_UNIT));
-    Preconditions.checkNotNull(configuration.getString(PROPERTIES));
     Preconditions.checkArgument(configuration.containsKey(ALERT_ID) || configuration.containsKey(ALERT_NAME));
     if (!configuration.containsKey(ALERT_ID)) {
       Preconditions.checkNotNull(configuration.getString(ALERT_TO));
     }
 
+    // check if duplicate name exists
+    List<AnomalyFunctionDTO> duplicateFunctions = anoomalyFunctionDAO.findWhereNameLike(configuration.getString(FUNCTION_NAME));
```

Comment:
Should we just create a method to find the name that matches new function's name?

Reply:
Fixed